### PR TITLE
Feature/ia micro services

### PR DESCRIPTION
**Pull request by tamaragovorcin**

### CODE REVIEW

File: IntegrationWithPharmacies/EPrescriptionApi/Model/RegistrationInPharmacy.cs [hunk 1]
```diff
@@ -15,14 +12,14 @@ public class RegistrationInPharmacy : Entity
 
         public RegistrationInPharmacy() : base() { }
 
-        public RegistrationInPharmacy(int id, int idPharmacy, String apiKeyPharmacy,String pharmacyName, String pharmacyTown) : base(id)
+        public RegistrationInPharmacy(int id, int idPharmacy, String apiKeyPharmacy, String pharmacyName, String pharmacyTown) : base(id)
         {
             PharmacyId = idPharmacy;
             ApiKey = apiKeyPharmacy;
             Name = pharmacyName;
             Town = pharmacyTown;
```
Comment (velicko-tam) [inline IntegrationWithPharmacies/EPrescriptionApi/Model/RegistrationInPharmacy.cs:20]: Change parameter names to match for example: PharmacyId = pharmacyId
Apply  this logic in all constructors

File: IntegrationWithPharmacies/EPrescriptionApi/Model/RegistrationInPharmacy.cs [hunk 2]
```diff
@@ -15,14 +12,14 @@ public class RegistrationInPharmacy : Entity
...
         }
-          public RegistrationInPharmacy(int idPharmacy, String apiKeyPharmacy,String pharmacyName, String pharmacyTown) 
+        public RegistrationInPharmacy(int idPharmacy, String apiKeyPharmacy, String pharmacyName, String pharmacyTown)
         {
             PharmacyId = idPharmacy;
```
Comment (velicko-tam) [inline IntegrationWithPharmacies/EPrescriptionApi/Model/RegistrationInPharmacy.cs:24]: Change parameter names to match for example: PharmacyId = pharmacyId
Apply  this logic in all constructors

File: IntegrationWithPharmacies/EPrescriptionApi/Service/MedicineAvailabilityTable.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+namespace EPrescriptionApi.Service
+{
+    public class MedicineAvailabilityTable
+    {
+        public MedicineAvailabilityTable() { }
+        public List<MedicineName> FormMedicineAvailability(string availability)
+        {
+            if (availability.Length > 5) return GetMedicineAvailabilityTable(availability);
+            return null;
```
Comment (velicko-tam) [inline IntegrationWithPharmacies/EPrescriptionApi/Service/MedicineAvailabilityTable.cs:14]: Write this as a ternary operator

File: IntegrationWithPharmacies/EPrescriptionApi/Service/SmptServerService.cs [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        public SmptServerService() { }
+
+        public void SendEMailNotification(String filePath)
+        {
+            try { SendMail(filePath); }
+            catch (SmtpException exception) { Console.WriteLine(exception.Message); }
+        }
+        private static void SendMail(String filePath)
+        {
+            SmtpClient SmptServer = FormSmptServerInformation();
```
Comment (velicko-tam) [inline IntegrationWithPharmacies/EPrescriptionApi/Service/SmptServerService.cs:18]: Change SmptServer object name to start with a lower letter "smptServer".

File: IntegrationWithPharmacies/EPrescriptionApi/Service/SmptServerService.cs [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+            SmptServer.Send(CreateMailMessage(filePath));
+        }
+   
+        private static SmtpClient FormSmptServerInformation()
+        {
+            SmtpClient SmptServer = new SmtpClient("smtp.gmail.com", 587);
```
Comment (velicko-tam) [inline IntegrationWithPharmacies/EPrescriptionApi/Service/SmptServerService.cs:24]: Change SmptServer object name to start with a lower letter "smptServer".

File: IntegrationWithPharmacies/EPrescriptionApi/Utility/HttpRequests.cs [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+using System.Threading.Tasks;
+
+namespace EPrescriptionApi.Utility
+{
+    public class HttpRequests
+    {
+        private static readonly string medicineInformationUrl = Startup.Configuration["MedicineInformationApi"];
+        private static readonly string usersServiceUrl = Startup.Configuration["UserMicroServiceApi"];
+        private static readonly string pharmacyRegistrationUrl = Startup.Configuration["PharmacyRegistrationApi"];
+
```
Comment (velicko-tam) [inline IntegrationWithPharmacies/EPrescriptionApi/Utility/HttpRequests.cs:19]: Change names of private static strings to start with upper letter.

File: IntegrationWithPharmacies/IntegrationWithPharmacies/Services/RabbitMQService.cs [hunk 7]
```diff
@@ -14,6 +14,8 @@ namespace IntegrationWithPharmacies
 {
     public class RabbitMQService : BackgroundService
     {
+        //private static readonly string usersServiceUrl = Startup.Configuration["UserMicroServiceApi"];
```
Comment (velicko-tam) [inline IntegrationWithPharmacies/IntegrationWithPharmacies/Services/RabbitMQService.cs:17]: Remove unnecessary comment

File: IntegrationWithPharmacies/PharmacyRegistrationApi/Model/RegistrationInPharmacy.cs [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+
+        public RegistrationInPharmacy() : base() { }
+
+        public RegistrationInPharmacy(int id, int idPharmacy, String apiKeyPharmacy, String pharmacyName, String pharmacyTown) : base(id)
+        {
+            PharmacyId = idPharmacy;
+            ApiKey = apiKeyPharmacy;
+            Name = pharmacyName;
+            Town = pharmacyTown;
```
Comment (velicko-tam) [inline IntegrationWithPharmacies/PharmacyRegistrationApi/Model/RegistrationInPharmacy.cs:20]: Change parameter name to match left side for example it should be "PharmacyId = pyharmacyId"

File: IntegrationWithPharmacies/PharmacyRegistrationApi/Model/RegistrationInPharmacy.cs [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+        }
+        public RegistrationInPharmacy(int idPharmacy, String apiKeyPharmacy, String pharmacyName, String pharmacyTown)
+        {
+            PharmacyId = idPharmacy;
+            ApiKey = apiKeyPharmacy;
+            Name = pharmacyName;
+            Town = pharmacyTown;
```
Comment (velicko-tam) [inline IntegrationWithPharmacies/PharmacyRegistrationApi/Model/RegistrationInPharmacy.cs:27]: Change parameter name to match left side for example it should be "PharmacyId = pharmacyId"
Apply in other constructors.

File: IntegrationWithPharmacies/TenderApi/Dto/TenderDto.cs [hunk 10]
```diff
@@ -1,18 +1,18 @@
...
+    public class TenderDto
+    {
+        public DateTime ActiveUntil { get; set; }
+        public bool Closed { get; set; }
+
+        public TenderDto() { }
+
+        public TenderDto(DateTime date, bool closed)
+        {
+            ActiveUntil = date;
```
Comment (velicko-tam) [inline IntegrationWithPharmacies/TenderApi/Dto/TenderDto.cs:14]: Change parameter name to match left side for example it should be "ActiveUntil = activeUntil".
You could also rename "ActiveUntil" into "ExpirationDate" if it makes more sense in the given context.

File: IntegrationWithPharmacies/TenderApi/Model/RegistrationInPharmacy.cs [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+
+
+        public RegistrationInPharmacy() : base() { }
+
+        public RegistrationInPharmacy(int id, int idPharmacy, String apiKeyPharmacy, String pharmacyName, String pharmacyTown) : base(id)
+        {
+            PharmacyId = idPharmacy;
+            ApiKey = apiKeyPharmacy;
+            Name = pharmacyName;
+            Town = pharmacyTown;
```
Comment (velicko-tam) [inline IntegrationWithPharmacies/TenderApi/Model/RegistrationInPharmacy.cs:20]: Change parameter name to match left side for example it should be "PharmacyId = pyharmacyId"
Apply to all constructors.

File: IntegrationWithPharmacies/TenderApi/Model/RegistrationInPharmacy.cs [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+        }
+        public RegistrationInPharmacy(int idPharmacy, String apiKeyPharmacy, String pharmacyName, String pharmacyTown)
+        {
+            PharmacyId = idPharmacy;
+            ApiKey = apiKeyPharmacy;
+            Name = pharmacyName;
+            Town = pharmacyTown;
```
Comment (velicko-tam) [inline IntegrationWithPharmacies/TenderApi/Model/RegistrationInPharmacy.cs:27]: Change parameter name to match left side for example it should be "PharmacyId = pharmacyId".
Apply to all constructors.

File: IntegrationWithPharmacies/TenderApi/Model/Tender.cs [hunk 13]
```diff
@@ -1,24 +1,23 @@
...
+    {
+        public DateTime ActiveUntil { get; set; }
+        public bool Closed { get; set; }
+
+        public Tender() : base() { }
+
+        public Tender(int id, DateTime date, bool closed) : base(id)
+        {
+            ActiveUntil = date;
+            Closed = closed;
```
Comment (velicko-tam) [inline IntegrationWithPharmacies/TenderApi/Model/Tender.cs:15]: Change parameter name to match left side for example it should be "ActiveUntil= activeUntil".
You could also try changing name of "ActiveUntil" into "ExpirationDate" if it makes sense in the given context.

File: IntegrationWithPharmacies/TenderApi/Model/Tender.cs [hunk 14]
```diff
@@ -1,24 +1,23 @@
...
+        }
+        public Tender(DateTime date, bool closed)
+        {
+            ActiveUntil = date;
```
Comment (velicko-tam) [inline IntegrationWithPharmacies/TenderApi/Model/Tender.cs:19]: Change parameter name to match left side for example it should be "ActiveUntil= activeUntil".
You could also try changing name of "ActiveUntil" into "ExpirationDate" if it makes sense in the given context.

File: IntegrationWithPharmacies/TenderApi/Service/SmptServerService.cs [hunk 15]
```diff
@@ -61,7 +61,7 @@ private static void SendMail(String filePath, String type)
             SmtpClient SmptServer = FormSmptServerInformation();
```
Comment (velicko-tam) [inline IntegrationWithPharmacies/TenderApi/Service/SmptServerService.cs:61]: Change name of SmtpClient object "SmptServer" into "smptServer" so it starts with a lower letter.

File: IntegrationWithPharmacies/TenderApi/Service/SmptServerService.cs [hunk 16]
```diff
@@ -61,7 +61,7 @@ private static void SendMail(String filePath, String type)
...
             SmptServer.Send(CreateMailMessage(filePath, type));
         }
-   
+
         private static SmtpClient FormSmptServerInformation()
         {
             SmtpClient SmptServer = new SmtpClient("smtp.gmail.com", 587);
```
Comment (velicko-tam) [inline IntegrationWithPharmacies/TenderApi/Service/SmptServerService.cs:67]: Change name of SmtpClient object "SmptServer" into "smptServer" so it starts with a lower letter.

File: IntegrationWithPharmacies/UrgentMedicineOrderApi/Model/Entity.cs [hunk 17]
```diff
@@ -0,0 +1,13 @@
...
+namespace UrgentMedicineOrderApi.Model
+{
+    public class Entity
+    {
+        public Entity() { }
+        public Entity(int id)
+        {
+            this.id = id;
+        }
+        public int id { get; set; }
```
Comment (velicko-tam) [inline IntegrationWithPharmacies/UrgentMedicineOrderApi/Model/Entity.cs:11]: Change name of property "id" into "Id".
Remove "this" from constructor and change it into "Id = id".

File: IntegrationWithPharmacies/UrgentMedicineOrderApi/Repository/UrgentMedicineOrderRepository.cs [hunk 18]
```diff
@@ -1,28 +1,28 @@
...
+﻿using System.Collections.Generic;
+using System.Linq;
+using UrgentMedicineOrderApi.DbContextModel;
+using UrgentMedicineOrderApi.Model;
+
+namespace UrgentMedicineOrderApi.Repository
+{
+   public class UrgentMedicineOrderRepository : IUrgentMedicineOrderRepository
+    {
+        private MyDbContext dbContext;
```
Comment (velicko-tam) [inline IntegrationWithPharmacies/UrgentMedicineOrderApi/Repository/UrgentMedicineOrderRepository.cs:10]: Change name of MyDbContext property "dbContext" into "DbContext" so it starts with an upper letter.
Remove "this" in constructors and make it to be "DbContext = dbContext".
Remove  "this.dbContext" were possible and put "DbContext".

File: IntegrationWithPharmacies/UrgentMedicineOrderApi/Service/UrgentOrderService.cs [hunk 19]
```diff
@@ -1,82 +1,82 @@
...
+                HttpService.SendUrgentOrder(CreateOrder(order));
+                return true;
+            }
+            catch (Exception e) { return false; }
+        }
+        public Boolean SendOrderGrpc(UrgentMedicineOrder order)
+        {
+            try
+            {
+                string response = new ClientScheduledService().SendMessage(CreateOrder(order)).Result;
```
Comment (velicko-tam) [inline IntegrationWithPharmacies/UrgentMedicineOrderApi/Service/UrgentOrderService.cs:60]: Replace string response with a "_"

**Review by velicko-tam — CHANGES_REQUESTED** [review]
PR is not approved. Marked parts should be fixed.

**Review by velicko-tam — APPROVED** [review]
PR is approved. Clean code principles were implemented.

**Review by trki293 — APPROVED** [review]
PR is approved. Good Clean code practices are followed.

**Pull request merged**